### PR TITLE
[5.3] Skip to amount to buy when region is selected

### PR DESCRIPTION
**Pull request by wachunei**
~~> This PR is using base `release/5.2.0`. When that branch is merged to `main`, this should be updated to point to `main` as base.~~

This PR adds the skipping screen feature for on ramp aggregator.
- When user has not previously selected a region, they will land on the Region screen (current behavior).
- When user has previously selected a region, they will land on Amount to buy screen.
  - There is a small chance that the selected region changes to be unsupported, when that happens we redirect to Region screen and remove that selection.

### CODE REVIEW

File: app/components/Nav/Main/MainNavigator.js [hunk 1]
```diff
@@ -441,6 +441,11 @@ const FiatOnRampAggregator = () => (
         component={Region}
         options={{ animationEnabled: false }}
       />
+      <Stack.Screen
```
Comment (alaahd) [inline app/components/Nav/Main/MainNavigator.js:444]: I do understand we need some routes here with disabled animations, those routes will still point to the same components but will be carrying a different behavior "disabled animation" I was thinking if you can avoid the creation of additional stacks by setting option dynamically before you `navigate` or `reset` some thing like : 
```suggestion
     navigation.setOptions({
      animationEnabled: false
    })
```
I don't see any major issue with this approach just asking if we can optimize one step further.

Comment (alaahd) [reply app/components/Nav/Main/MainNavigator.js:444]: As long as there is a way to dynamically set options specifically without affecting the global behavior.

File: app/components/UI/FiatOnRampAggregator/Views/GetStarted.tsx [hunk 2]
```diff
@@ -88,12 +93,24 @@ const GetStarted: React.FC = () => {
 
   useEffect(() => {
     if (getStarted) {
-      navigation.reset({
-        index: 0,
-        routes: [{ name: Routes.FIAT_ON_RAMP_AGGREGATOR.REGION_HAS_STARTED }],
-      });
+      if (selectedRegion) {
```
Comment (alaahd) [inline app/components/UI/FiatOnRampAggregator/Views/GetStarted.tsx:96]: Do you think we should also check for the selected payment method in addition to selectedRegion?

[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/Views/GetStarted.tsx:96]: This is not part of the task

File: app/components/Nav/Main/MainNavigator.js (hunk 1 above)
Comment (alaahd) [reply app/components/Nav/Main/MainNavigator.js:449]: Ok perfect. this was already a non-blocking feedback.

**Review by alaahd — APPROVED** [review]
LGTM!

**Review by alaahd — APPROVED** [review]

**Comment by Cal-L** [comment]
Let's account for iPhones with notches on the bottom

image

**Review by Cal-L — APPROVED** [review]
Aside from my comment, LGTM!

**Pull request merged**